master/types.html

```diff
@@ -207,6 +207,477 @@ <h2 id="Precision">Real number precision</h2>
+
+<p>Some numeric attribute and property values have restricted ranges.
+Other values will be restricted by the capabilities of the device.
+If not otherwise specified,
+the user agent shall defer any out-of-range error checking until
+as late as possible in the rendering process.
+This is particularly important for device limitations,
+as compound operations
+might produce intermediate values which are out-of-range but
+final values which are within range.</p>
+
+<h2 id="SVGDOMOverview">SVG DOM overview</h2>
+
+<p>Conforming <a>SVG viewers</a> or <a>SVG interpreters</a>
+that support <a>script execution</a>
+must implement SVG DOM interfaces as defined throughout this specification,
+with the specific requirements and dependencies listed in this section.
+</p>
+
+<div class="annotation svg2-requirement">
+  <table>
+    <tr>
+      <th>SVG 2 Requirement:</th>
+      <td>Improve the DOM.</td>
+    </tr>
+    <tr>
+      <th>Resolution:</th>
+      <td><a href="http://www.w3.org/2011/10/27-svg-irc#T18-35-49">We will generally improve the SVG DOM for SVG 2.</a></td>
+    </tr>
+    <tr>
+      <th>Purpose:</th>
+      <td>Help authors use the SVG DOM by making it less Java-oriented.</td>
+    </tr>
+    <tr>
+      <th>Owner:</th>
+      <td>Cameron (<a href="http://www.w3.org/Graphics/SVG/WG/track/actions/3273">ACTION-3273</a>)</td>
+    </tr>
+    <tr>
+      <th>Note:</th>
+      <td>See <a href="http://www.w3.org/Graphics/SVG/WG/wiki/SVG_2_DOM">SVG 2 DOM</a> Wiki page.</td>
+    </tr>
+  </table>
+</div>
+
+<div class="annotation svg2-requirement">
+  <table>
+    <tr>
+      <th>SVG 2 Requirement:</th>
+      <td>Improve the SVG path DOM APIs.</td>
+    </tr>
+    <tr>
+      <th>Resolution:</th>
+      <td><a href="http://www.w3.org/2011/10/27-svg-irc#T18-23-23">We will improve the SVG path DOM APIs in SVG 2.</a></td>
+    </tr>
+    <tr>
+      <th>Purpose:</th>
+      <td>Clean up SVGPathSegList interface, and possibly share an API with Canvas.</td>
+    </tr>
+    <tr>
+      <th>Owner:</th>
+      <td>Cameron (no action)</td>
+    </tr>
+  </table>
+</div>
+
+
+<h3 id="SVGDOMDependencies">Dependencies for SVG DOM support</h3>
+
+<p class="ready-for-wider-review">The SVG DOM is defined in terms of <a href="https://heycam.github.io/webidl/">Web IDL</a>
+interfaces. All IDL fragments in this specification must be interpreted as
+required for <a href="https://heycam.github.io/webidl/#dfn-conforming-set-of-idl-fragments">conforming IDL fragments</a>,
+as described in the Web IDL specification. [<a href="refs.html#ref-webidl">WebIDL</a>]</p>
+
+<p>The SVG DOM builds upon a number of DOM specifications.  In particular:</p>
+
+<ul>
+  <li>The SVG DOM requires, at a minimum, complete support for
+  <a href="https://www.w3.org/TR/dom/">DOM4</a>;
+  SVG software should implement
+  the <a href="https://dom.spec.whatwg.org/">latest DOM standard</a> 
+  wherever possible
+  [<a href="refs.html#ref-dom">DOM</a>]</li>
```

Comment:
@svgeesus said:

> Removing W3C DOM4 from refs was a deliberate change. We should refer to DOM LS instead.

That sounds reasonable to me, but we can't really say they "must" implement the absolute latest spec, since there's often a lag between stabilizing a new spec feature & getting an implementation out.

Any suggestions for handling it better? "must support all aspects that were included in the DOM standard as of the publication of this specification and which haven't been deprecated or removed since"?

Reply:
I don't have any suggestions for handling it better. Saying that implementers must implement something that has been dropped from LS will make us unpopular.

Reply:
Ok. That sort of matches how I'm handling the instability of CSSOM, anyway.